idunn/blocks/recycling.py

```diff
@@ -0,0 +1,100 @@
+from datetime import datetime, timedelta
+from typing import ClassVar, List
+from enum import Enum
+import requests
+from pydantic import BaseModel, validator
+from pytz import UTC
+
+from idunn import settings
+from .base import BaseBlock
+
+
+def is_poi_in_finistere(poi):
+    admins = poi.get_raw_admins()
+    return any(
+        any(c.get("name") == "ISO3166-2" and c.get("value") == "FR-29" for c in a.get("codes", []))
+        for a in admins
+    )
+
+
+class ContainerType(str, Enum):
+    paper = "paper"
+    glass = "glass"
+    unknown = "unknown"
+
+
+class RecyclingContainer(BaseModel):
+    updated_at: datetime
+    filling_level: int
+    type: ContainerType
+
+    @validator("updated_at")
+    def validate_datetime(cls, dt):
+        if dt.tzinfo is None:
+            return UTC.localize(dt)
+        return dt.as_timezone(UTC)
+
+
+class RecyclingBlock(BaseBlock):
+    BLOCK_TYPE: ClassVar = "recycling"
+
+    containers: List[RecyclingContainer]
+
+    @classmethod
+    def from_es(cls, place, lang):
+        if not settings.get("RECYCLING_SERVER_URL"):
+            # Data source is not configured
+            return None
+
+        if place.PLACE_TYPE != "poi":
+            return None
+
+        if place.get_class_name() != "recycling":
+            return None
+
+        if not is_poi_in_finistere(place):
```

Comment:
Can't you make the three conditions into one?

Reply:
These conditions will change very soon, and should probably be set in the configuration (for example to allow multiple zones where this block should be enabled).

I will keep them for now, waiting for more specifications.